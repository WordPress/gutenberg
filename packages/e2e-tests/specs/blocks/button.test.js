/**
 * WordPress dependencies
 */
import {
	insertBlock,
	getEditedPostContent,
	createNewPost,
} from '@wordpress/e2e-test-utils';

describe( 'Button', () => {
	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'has focus on button content', async () => {
		await insertBlock( 'Button' );
		await page.keyboard.type( 'Content' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can jump focus back & forth', async () => {
		await insertBlock( 'Button' );
		await page.keyboard.type( 'WordPress' );
		await page.keyboard.press( 'Tab' );
		await page.keyboard.type( 'https://wordpress.org' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
