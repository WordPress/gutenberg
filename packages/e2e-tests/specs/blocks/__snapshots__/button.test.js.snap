// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button can jump focus back & forth 1`] = `
"<!-- wp:button -->
<div class=\\"wp-block-button\\"><a class=\\"wp-block-button__link\\" href=\\"https://wordpress.org\\">WordPress</a></div>
<!-- /wp:button -->"
`;

exports[`Button has focus on button content 1`] = `
"<!-- wp:button -->
<div class=\\"wp-block-button\\"><a class=\\"wp-block-button__link\\">Content</a></div>
<!-- /wp:button -->"
`;
